core/src/main/java/org/apache/calcite/sql/validate/BlockScope.java

```diff
@@ -52,14 +57,19 @@ public BlockScope(SqlValidatorScope parent, SqlLabeledBlock block) {
 
   @Override public void resolve(List<String> names, SqlNameMatcher nameMatcher,
       boolean deep, Resolved resolved) {
-    if (block.label == null) {
+    if (names.size() != 1) {
```

Comment:
The condition names are only valid as a SimpleIdentifier which has 1 name. The other cases would be passed along and handled by a parent scope.

Reply:
Let's use SqlIdentifier's `isSimple()` function in that case to make our reasoning for this check more explicit.

Reply:
The resolve method is passing a list of String values so it wouldn't have access in this method to isSimple(). Otherwise, I could add a comment here clarifying why it ignores the other cases

Reply:
Ah I see. Makes sense then. Just a comment sounds good then.